api/apiroutes.go

```diff
@@ -1454,8 +1459,12 @@ func (c *appContext) addressIoCsv(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	c.statusMtx.RLock()
+	statusHeight := c.Status.Height
+	c.statusMtx.RUnlock()
+
 	filename := fmt.Sprintf("address-io-%s-%d-%s.csv", address,
-		c.Status.GetHeight(), strconv.FormatInt(time.Now().Unix(), 10))
+		statusHeight, strconv.FormatInt(time.Now().Unix(), 10))
```

Comment:
This is actually why I avoid adding named mutexes that apply to a specific field of the struct.  Having the Status type have it's own mutex makes ensures the mutex is only used for Status accesses.  Plus the `GetHeight()` method does the locking for you, and there's no temp variable.  All around it's cleaner IMO.

Reply:
The fields are exported so the API allows you to bypass the function calls that keep data integrity.  Perhaps a third solution is needed :)

Reply:
True, there is ample opportunity for messing with the data the way it is set up.  Unexporting the fields and making a setter method is probably the third method then.